### ISSUE TITLE
Update sidebar example to match audit results

### ISSUE THREAD

**Issue by ethanwinters** [carbon-design-system/carbon-ai-chat#526]
Update demo site and make an example for the recommended way to do the sidebar interaction: https://www.figma.com/design/6wO5ZNTAxBII6MoVyRIYqW/Audit-documentation?node-id=100-21&p=f&t=T71cjHBNQXORLUDk-0

Also, be wary of previous a11y issues on the demo site and make sure they are resolved in this rework:
  
___

Figure 1 The sidebar icon has some issues
The expand/collapse icon [shows up after the dropdowns in the tab order](https://ibm.box.com/s/fzcrm3plks8bnfz0d4rg9hwe2su0vsog) (focus goes from product title to inputs then up to icon). I’m assuming this is just a result of the demo, but I am explicitly noting that if unchanged, this would verge on a failure of Focus Order; since it is in the header, it should precede items on the page.

The expand icon is not showing a title on hover or focus. Normally it would in Carbon. There is no tooltip on the AI icon either. I noticed that when the sidebar icon is expanded, it does have a tooltip (see Figure 2). So, both missing tooltips are bugs to be addressed.

Figure 2 A properly displaying tooltip
After the Sidebar is opened and then collapsed, a [number of tab stops occur with no visible focus visible](https://ibm.box.com/s/g7hbi72zknsq8x9gvgqz9z2qo924uzeb). I suspect what is happening is that the sidebar is not being properly hidden programmatically and so is still showing up in the DOM and its tab elements continue to receive KB focus, even though nothing is visible. This is a sev 2 fail of Focus Visible and should be addressed.
